fl4health/clients/basic_client.py

```diff
@@ -748,7 +753,10 @@ def _validate_or_test(
         metric_manager.clear()
         loss_meter.clear()
         with torch.no_grad():
-            for input, target in maybe_progress_bar(loader, self.progress_bar):
+            for i, (input, target) in enumerate(maybe_progress_bar(loader, self.progress_bar)):
```

Comment:
So this approach works great for the nnunet client where the dataloaders randomly sample from the set. But if we have a traditional dataloader that runs through the val set, then will setting a max number of steps here that doesn't cover the whole validation set mean that we're basically not using part of the validation set? If so do we care? Is it worth trying to add logic that ensures the whole validation set is used?

Reply:
I think it's a lever you would only want to turn on if you explicitly don't want to use the entire validation set (ex: for performance reasons). In all other instances, you would not specify max_num_validation_steps and get the expected behaviour of going through entire validation set

Reply:
I will add some sort of log that warns that we are sampling from validation set when max_validation_steps is specified